cumulus_library/databases/base.py

```diff
@@ -120,6 +120,10 @@ def __init__(self, schema_name: str):
         # technology
         self.db_type = None
 
+    @abc.abstractmethod
+    def connect(self):
+        """Initiates connection configuration of the database"""
```

Comment:
If we wanted to be Extremely Python :tm:, we would probably write this abstract class as a context manager, so you could do:
```
with AthenaDatabase(...) as db:
    # use db
```

To automatically get connect/close support (and make sure that callers don't forget one of the calls, even when an exception happens).

But not necessary right now and can often be a little annoying to reorganize code to support that style (but does have some benefits - mainly the exception handling).